manager/state/raft/raft.go

```diff
@@ -342,6 +398,9 @@ func (n *Node) JoinAndStart(ctx context.Context) (err error) {
 	}
 
 	// join to existing cluster
+	if n.opts.Addr == "" {
```

Comment:
You need to bind first if you are trying to join a cluster, but if you're starting a new cluster then you hit the `n.opts.JoinAddr == ""` branch and having an address is not required.